### ISSUE TITLE
Fluent Bit sometimes dies with SIGABRT

### ISSUE THREAD

**Issue by LionelCons** [fluent/fluent-bit#2403]
After deploying Fluent Bit on several machines and let it run for some hours, I saw two occurrences the daemon crashing with SIGABRT.

I managed to capture a core dump for the second crash. It contains:
```
Program terminated with signal 6, Aborted.
#0  0x00007fd5d7624387 in raise () from /lib64/libc.so.6
(gdb) bt
#0  0x00007fd5d7624387 in raise () from /lib64/libc.so.6
#1  0x00007fd5d7625a78 in abort () from /lib64/libc.so.6
#2  0x00000000004351c6 in flb_signal_handler ()
#3  <signal handler called>
#4  0x00000000004b612a in __mk_list_del ()
#5  0x00000000004b6161 in mk_list_del ()
#6  0x00000000004b6a27 in destroy_conn ()
#7  0x00000000004b6da8 in cb_upstream_conn_ka_dropped ()
#8  0x00000000004b0ee6 in flb_engine_start ()
#9  0x00000000004364b6 in flb_main ()
#10 0x000000000043650a in main ()
```

I don't know how to reproduce this but maybe the stack trace is enough to find where the problem might be?

**Comment by edsiper** [comment]
I suspect what can be, troubleshooting.

did you compile your own version ? I am wondering why there is no debug symbols.

**Comment by LionelCons** [comment]
Yes, it is my own compiled version.

It happened once more this morning with the exact same trace.

**Comment by edsiper** [comment]
when configuring please use the flag in CMake command ```-DFLB_DEV=on```, that one will enable debug symbols.

**Comment by LionelCons** [comment]
One more crash, this time with a bit more information:

```
(gdb) bt
#0  0x00007fc5527a7387 in raise () from /lib64/libc.so.6
#1  0x00007fc5527a8a78 in abort () from /lib64/libc.so.6
#2  0x00000000004352a6 in flb_signal_handler (signal=11) at fluent-bit-1.5.2/src/fluent-bit.c:404
#3  <signal handler called>
#4  0x00000000004b6345 in __mk_list_del (prev=0x0, next=0x0) at fluent-bit-1.5.2/lib/monkey/include/monkey/mk_core/mk_list.h:87
#5  0x00000000004b637c in mk_list_del (entry=0x7fc550643778) at fluent-bit-1.5.2/lib/monkey/include/monkey/mk_core/mk_list.h:93
#6  0x00000000004b6c42 in destroy_conn (u_conn=0x7fc550643700) at fluent-bit-1.5.2/src/flb_upstream.c:263
#7  0x00000000004b6fc3 in cb_upstream_conn_ka_dropped (data=0x7fc550643700) at fluent-bit-1.5.2/src/flb_upstream.c:379
#8  0x00000000004b1101 in flb_engine_start (config=0x7fc550610f40) at fluent-bit-1.5.2/src/flb_engine.c:607
#9  0x0000000000436596 in flb_main (argc=4, argv=0x7ffdfd9a9438) at fluent-bit-1.5.2/src/fluent-bit.c:1035
#10 0x00000000004365ea in main (argc=4, argv=0x7ffdfd9a9438) at fluent-bit-1.5.2/src/fluent-bit.c:1048
```

**Comment by LionelCons** [comment]
Crashes continue with always the same stack trace :-(

I also noticed something weird in the program's output (last 3 lines):
```
Aug 06 00:52:25 localhost fluent-bit[18446]: [engine] caught signal (SIGSEGV)
Aug 06 00:52:25 localhost fluent-bit[18446]: #0  0x4b6345            in  __mk_list_del() at lib/monkey/include/monkey/mk_core/mk_list.h:87
Aug 06 00:52:25 localhost fluent-bit[18446]: #1  0x4b637b            in  mk_list_del() at lib/monkey/include/monkey/mk_core/mk_list.h:93
Aug 06 00:52:25 localhost fluent-bit[18446]: #2  0x4b6c41            in  destroy_conn() at src/flb_upstream.c:263
Aug 06 00:52:25 localhost fluent-bit[18446]: #3  0x4b6fc2            in  cb_upstream_conn_ka_dropped() at src/flb_upstream.c:379
Aug 06 00:52:25 localhost fluent-bit[18446]: #4  0x4b1100            in  flb_engine_start() at src/flb_engine.c:607
Aug 06 00:52:25 localhost fluent-bit[18446]: #5  0x436595            in  flb_main() at src/fluent-bit.c:1035
Aug 06 00:52:25 localhost fluent-bit[18446]: #6  0x4365e9            in  main() at src/fluent-bit.c:1048
Aug 06 00:52:25 localhost fluent-bit[18446]: #7  0x7f5f9032d554      in  ???() at ???:0
Aug 06 00:52:25 localhost fluent-bit[18446]: #8  0x434189            in  ???() at ???:0
Aug 06 00:52:25 localhost fluent-bit[18446]: #9  0xffffffffffffffff  in  ???() at ???:0
```

This is maybe an unrelated bug in libbacktrace...

**Comment by edsiper** [comment]
Fixed in #2531
